### PR TITLE
Refactor collective model to typescript

### PR DESCRIPTION
**Pull request by hdiniz**
Related https://github.com/opencollective/opencollective/issues/5893
Might be better to review by following the individual commits on 'Collective.ts`.

### CODE REVIEW

File: server/lib/merge-accounts.ts [hunk 1]
```diff
@@ -80,7 +78,7 @@ export const simulateMergeAccounts = async (
 
   const addCountsToSummary = counts => {
     Object.entries(counts).forEach(([key, count]) => {
-      if (count > 0) {
+      if ((count as number) > 0) {
         addLineToSummary(`  - ${key}: ${count}`);
       }
     });
```
Comment (Betree) [inline server/lib/merge-accounts.ts:84]: Wouldn't it be better to type it in the function parameter? `as` kind of bypasses type checks

```es6
const addCountsToSummary = (counts: Record<string, number>) => {
```

Comment (Betree) [reply server/lib/merge-accounts.ts:84]: Note that this one doesn't seem to original from this PR, I'm curious why we haven't seen it before

File: test/server/lib/merge-accounts.test.ts [hunk 2]
```diff
@@ -93,7 +93,7 @@ describe('server/lib/merge-accounts', () => {
 
   before(async function () {
     // This test can take some time to setup on local env
-    this.timeout(45000);
+    this.timeout(90000);
```
Comment (Betree) [inline test/server/lib/merge-accounts.test.ts:96]: - https://github.com/opencollective/opencollective/issues/6533

File: test/test-helpers/fake-data.ts [hunk 3]
```diff
@@ -358,7 +369,7 @@ export const fakeExpense = async (expenseData: Record<string, unknown> = {}) =>
  * Creates a fake comment. All params are optionals.
  */
 export const fakeComment = async (
-  commentData: Record<string, unknown> = {},
+  commentData: Partial<InferCreationAttributes<Comment>> = {},
```
Comment (Betree) [inline test/test-helpers/fake-data.ts:372]: Nice job typing all these with `Partial<InferCreationAttributes<...>>` :+1:

**Review by Betree — APPROVED** [review]
Looking great! This is a massive improvement.

File: server/lib/merge-accounts.ts (hunk 1 above)
Comment (hdiniz) [reply server/lib/merge-accounts.ts]: Changed the CollectiveFieldsConfig type as merge-accounts, maybe it propagated to here.

**Pull request merged**